core/src/main/java/com/linecorp/armeria/server/encoding/EncodingServiceBuilder.java

```diff
@@ -120,12 +127,36 @@ public EncodingServiceBuilder minBytesToForceChunkedEncoding(int minBytesToForce
         return this;
     }
 
+    /**
+     * Sets the specified {@link Encoding}s for the {@link EncodingService} should enable/support.
+     * The default is {@link EncodingServiceBuilder#ALL_ENCODINGS}.
+     *
+     * @param encodings the {@link Encoding}s to enable
+     */
+    public EncodingServiceBuilder enabledEncodings(Encoding... encodings) {
```

Comment:
Question) Given that `StreamEncoderFactories` already exists, I'm wondering if it makes sense to reuse it.

```suggestion
    public EncodingServiceBuilder enabledEncoderFactories(Encoding... encodings) {
```

Reply:
We can make it public though as you suggested: https://github.com/line/armeria/blob/882a63cd27857b06c61bfc8a02a1ef0a995ed698/core/src/main/java/com/linecorp/armeria/internal/common/encoding/StreamEncoderFactory.java#L25

Reply:
You are right. We didn't expose it as a public API since there is no API to customize it. I prefer exposing an interface instead of an enum so that users can implement their own encoders.

https://github.com/line/armeria/blob/5959c832e0c3f126b2560f8a36a5bb3383e5bcc9/core/src/main/java/com/linecorp/armeria/internal/common/encoding/StreamEncoderFactory.java#L25

To fulfill the purpose of this PR, we also need to make `StreamEncoderFactory` a public APi.

Reply:
@jrhee17, @ikhoon pushed a solution